### ISSUE TITLE
Structured list example: can't select with keyboard

### ISSUE THREAD

**Issue by carmacleod** [carbon-design-system/carbon#7144]
If I click with the mouse on a row in the [Structured list example](https://carbon-components-react.netlify.app/iframe.html?id=structuredlist--selection), the little checkmark moves to the clicked row. There does not seem to be a way to do this with the keyboard.

**Comment by tw15egan** [comment]
I think this is blocked until we can refactor this component as the changes needed are breaking changes 

Rel https://github.com/carbon-design-system/carbon/issues/2820
Rel https://github.com/carbon-design-system/carbon/issues/5065
Rel https://github.com/carbon-design-system/carbon/issues/1937

**Comment by joshblack** [comment]
Closing as duplicate of #2820 which is in our WCAG 2.1 epic
